diff --git a/debian/generate-wisdom b/debian/generate-wisdom
index dc98d72..ef3dffa 100755
--- a/debian/generate-wisdom
+++ b/debian/generate-wisdom
@@ -9,12 +9,11 @@ echo "Benchmarking .. this will take a while." >&2
 
 # generate initial wisdom so that twopass implementations have something to work with
 echo "First pass: generating $WORKDIR/wisdom.initial" >&2
-/usr/lib/dump1090-fa/starch-benchmark -i 5 -o $WORKDIR/wisdom.initial magnitude_uc8 magnitude_uc8_aligned mean_power_u16 mean_power_u16_aligned
+starch-benchmark -i 5 -o $WORKDIR/wisdom.initial magnitude_uc8 magnitude_uc8_aligned mean_power_u16 mean_power_u16_aligned
 
 # generate the real wisdom
-echo "Second pass: generating $WORKDIR/wisdom.local" >&2
-/usr/lib/dump1090-fa/starch-benchmark -i 5 -r $WORKDIR/wisdom.initial -o $WORKDIR/wisdom.local
+echo "Second pass: generating $SNAP_COMMON/wisdom.local" >&2
+starch-benchmark -i 5 -r $WORKDIR/wisdom.initial -o $SNAP_COMMON/wisdom.local
 
-echo "Wisdom written to $WORKDIR/wisdom.local" >&2
-echo "Copy this file to /etc/dump1090-fa/wisdom.local" >&2
-echo "(and restart dump1090-fa) to start using it." >&2
+echo "Wisdom written to $SNAP_COMMON/wisdom.local" >&2
+echo "run 'snap restart adsb-box.dump1090' to start using it." >&2
diff --git a/public_html_merged/flags.js b/public_html_merged/flags.js
index fb40163..442ac1c 100644
--- a/public_html_merged/flags.js
+++ b/public_html_merged/flags.js
@@ -195,7 +195,7 @@ var ICAO_Ranges = [
         { start: 0x004000, end: 0x0043FF, country: "Zimbabwe", flag_image: "Zimbabwe.png" },
 
         { start: 0xF00000, end: 0xF07FFF, country: "ICAO (temporary assignments)", flag_image: "blank.png" },
-        { start: 0x899000, end: 0x8993FF, country: "ICAO (special use)", flag_image: "blank.png" },
+        { start: 0x899000, end: 0x8993FF, country: "Taiwan", flag_image: "Taiwan.png" },
         { start: 0xF09000, end: 0xF093FF, country: "ICAO (special use)", flag_image: "blank.png" },
 
         // Block assignments mentioned in Chapter 9 section 4, at the end so they are only used if
diff --git a/public_html_merged/index.html b/public_html_merged/index.html
index 96c5fdc..9869701 100644
--- a/public_html_merged/index.html
+++ b/public_html_merged/index.html
@@ -393,7 +393,7 @@
 											<tbody>
 												<tr id="plane_row_template" class="plane_table_row hidden">
 													<td class="icaoCodeColumn">ICAO</td>
-													<td><img style="width: 20px; height: 12px" src="about:blank" alt="Flag"></td>
+													<td><img style="width: 20px; height: 12px" src="data:," alt="Flag"></td>
 													<td>FLIGHT</td>
 													<td>REGISTRATION</td>
 													<td>AIRCRAFT_TYPE</td>
