diff --git a/public_html/flags.js b/public_html/flags.js
index c813c09..4c8704f 100644
--- a/public_html/flags.js
+++ b/public_html/flags.js
@@ -195,7 +195,7 @@ var ICAO_Ranges = [
         { start: 0x004000, end: 0x0043FF, country: "Zimbabwe", flag_image: "Zimbabwe.png" },
 
         { start: 0xF00000, end: 0xF07FFF, country: "ICAO (temporary assignments)", flag_image: "blank.png" },
-        { start: 0x899000, end: 0x8993FF, country: "ICAO (special use)", flag_image: "blank.png" },
+        { start: 0x899000, end: 0x8993FF, country: "Taiwan", flag_image: "Taiwan.png" },
         { start: 0xF09000, end: 0xF093FF, country: "ICAO (special use)", flag_image: "blank.png" },
 
         // Block assignments mentioned in Chapter 9 section 4, at the end so they are only used if
diff --git a/dsp/generated/makefile.arm b/dsp/generated/makefile.arm
index 58eaf5b..5145c5b 100644
--- a/dsp/generated/makefile.arm
+++ b/dsp/generated/makefile.arm
@@ -22,7 +22,7 @@ STARCH_CFLAGS := -DSTARCH_MIX_ARM
 
 
 dsp/generated/flavor.armv7a_neon_vfpv4.o: dsp/generated/flavor.armv7a_neon_vfpv4.c dsp/impl/mean_power_u16.c dsp/impl/magnitude_power_uc8.c dsp/impl/magnitude_uc8.c dsp/impl/magnitude_sc16q11.c dsp/impl/magnitude_sc16.c
-	$(STARCH_COMPILE) $(STARCH_CFLAGS) -march=armv7-a+neon-vfpv4 -mfpu=neon-vfpv4 -ffast-math dsp/generated/flavor.armv7a_neon_vfpv4.c -o dsp/generated/flavor.armv7a_neon_vfpv4.o
+	$(STARCH_COMPILE) $(STARCH_CFLAGS) -march=armv7-a -mfpu=neon-vfpv4 -ffast-math dsp/generated/flavor.armv7a_neon_vfpv4.c -o dsp/generated/flavor.armv7a_neon_vfpv4.o
 
 dsp/generated/flavor.generic.o: dsp/generated/flavor.generic.c dsp/impl/mean_power_u16.c dsp/impl/magnitude_power_uc8.c dsp/impl/magnitude_uc8.c dsp/impl/magnitude_sc16q11.c dsp/impl/magnitude_sc16.c
 	$(STARCH_COMPILE) $(STARCH_CFLAGS)  dsp/generated/flavor.generic.c -o dsp/generated/flavor.generic.o
diff --git a/starch/example/generated/makefile.arm b/starch/example/generated/makefile.arm
index d312adc..c8c8811 100644
--- a/starch/example/generated/makefile.arm
+++ b/starch/example/generated/makefile.arm
@@ -22,10 +22,10 @@ STARCH_CFLAGS := -DSTARCH_MIX_ARM
 
 
 generated/flavor.armv7a_vfpv4.o: generated/flavor.armv7a_vfpv4.c impl/subtract_n.c
-	$(STARCH_COMPILE) $(STARCH_CFLAGS) -march=armv7-a+neon-vfpv4 -mfpu=neon-vfpv4 -ffast-math generated/flavor.armv7a_vfpv4.c -o generated/flavor.armv7a_vfpv4.o
+	$(STARCH_COMPILE) $(STARCH_CFLAGS) -march=armv7-a -mfpu=neon-vfpv4 -ffast-math generated/flavor.armv7a_vfpv4.c -o generated/flavor.armv7a_vfpv4.o
 
 generated/flavor.armv7a_vfpv3.o: generated/flavor.armv7a_vfpv3.c impl/subtract_n.c
-	$(STARCH_COMPILE) $(STARCH_CFLAGS) -march=armv7-a+neon-vfpv3 -mfpu=neon-vfpv3 -ffast-math generated/flavor.armv7a_vfpv3.c -o generated/flavor.armv7a_vfpv3.o
+	$(STARCH_COMPILE) $(STARCH_CFLAGS) -march=armv7-a -mfpu=neon-vfpv3 -ffast-math generated/flavor.armv7a_vfpv3.c -o generated/flavor.armv7a_vfpv3.o
 
 generated/flavor.generic.o: generated/flavor.generic.c impl/subtract_n.c
 	$(STARCH_COMPILE) $(STARCH_CFLAGS)  generated/flavor.generic.c -o generated/flavor.generic.o
diff --git a/Makefile.cpufeatures b/Makefile.cpufeatures
index 099714e..88774e2 100644
--- a/Makefile.cpufeatures
+++ b/Makefile.cpufeatures
@@ -21,6 +21,10 @@ ifeq ($(CPUFEATURES_ARCH), x86_64)
   CPUFEATURES_OBJS += cpu_features/src/cpuinfo_x86.o
 endif
 
+ifneq (,$(filter $(CPUFEATURES_ARCH), i386 i686))
+  CPUFEATURES_OBJS += cpu_features/src/cpuinfo_x86.o
+endif
+
 ifneq (,$(findstring arm,$(CPUFEATURES_ARCH)))
   CPUFEATURES_OBJS += cpu_features/src/cpuinfo_arm.o
 endif
