diff --git a/flightaware/client/cli.py b/flightaware/client/cli.py
index b9c52a4..9b3aee1 100644
--- a/flightaware/client/cli.py
+++ b/flightaware/client/cli.py
@@ -10,8 +10,13 @@ from mlat.client.coordinator import Coordinator
 from mlat.client.util import log, log_exc
 from mlat.client import options
 
+import sys
+from os.path import basename
+from prctl import set_name
 
 def _main():
+    set_name(basename(sys.argv[0]))
+
     # piaware will timestamp our log messages itself, suppress the normal logging timestamps
     mlat.client.util.suppress_log_timestamps = True
 
diff --git a/mlat/client/cli.py b/mlat/client/cli.py
index f6a79f2..d8535ff 100644
--- a/mlat/client/cli.py
+++ b/mlat/client/cli.py
@@ -26,8 +26,13 @@ from mlat.client.jsonclient import JsonServerConnection
 from mlat.client.coordinator import Coordinator
 from mlat.client import options
 
+import sys
+from os.path import basename
+from prctl import set_name
 
 def main():
+    set_name(basename(sys.argv[0]))
+
     parser = argparse.ArgumentParser(description="Client for multilateration.")
 
     options.make_inputs_group(parser)
diff --git a/setup.py b/setup.py
index 06d16f0..6849f38 100755
--- a/setup.py
+++ b/setup.py
@@ -31,6 +31,7 @@ setup(name='MlatClient',
           Extension('_modes',
                     include_dirs=["."],
                     sources=['_modes.c', 'modes_reader.c', 'modes_message.c', 'modes_crc.c'])],
+      install_requires=['python-prctl'],
       entry_points={
           'console_scripts': [
               'mlat-client = mlat.client.cli:main',
