diff --git a/package/fa_piaware_config.tcl b/package/fa_piaware_config.tcl
index fa0828d..31bc6f7 100644
--- a/package/fa_piaware_config.tcl
+++ b/package/fa_piaware_config.tcl
@@ -514,7 +514,7 @@ namespace eval ::fa_piaware_config {
 				set f [open $temppath "w" [expr {$fpstat(mode) & 0777}]]
 
 				# fix the ownership to match the old file
-				chown -fileid [list $fpstat(uid) $fpstat(gid)] $f
+				#chown -fileid [list $fpstat(uid) $fpstat(gid)] $f
 			}
 
 			# write the new data
diff --git a/programs/piaware/faup1090.tcl b/programs/piaware/faup1090.tcl
index 803404d..fc01909 100644
--- a/programs/piaware/faup1090.tcl
+++ b/programs/piaware/faup1090.tcl
@@ -63,6 +63,11 @@ proc connect_adsb_via_faup1090 {} {
 	}
 
 	# path to faup1090
+	if {[info exists ::env(SNAP)]} {
+		set path "$::env(SNAP)/usr/lib/piaware/helpers/faup1090"
+	} else {
+		set path "/usr/lib/piaware/helpers/faup1090"
+	}
 	set path [auto_execok "/usr/lib/piaware/helpers/faup1090"]
 	if {$path eq ""} {
 		logger "No faup1090 found at $path, 1090ES support disabled"
diff --git a/programs/piaware/mlat.tcl b/programs/piaware/mlat.tcl
index c511c60..eadd04c 100644
--- a/programs/piaware/mlat.tcl
+++ b/programs/piaware/mlat.tcl
@@ -15,7 +15,11 @@ set ::mlatRestartMillis 60000
 # UDP transport info
 set ::mlatUdpTransport {}
 # path to fa-mlat-client
-set ::mlatClientPath [auto_execok "/usr/lib/piaware/helpers/fa-mlat-client"]
+if {[info exists ::env(SNAP)]} {
+	set ::mlatClientPath [auto_execok "$::env(SNAP)/usr/lib/piaware/helpers/fa-mlat-client"]
+} else {
+	set ::mlatClientPath [auto_execok "/usr/lib/piaware/helpers/fa-mlat-client"]
+}
 # current mlat status for the statusfile, one of:
 #   not_enabled
 #   not_running
diff --git a/programs/piaware-config/main.tcl b/programs/piaware-config/main.tcl
index 388b957..61b4968 100644
--- a/programs/piaware-config/main.tcl
+++ b/programs/piaware-config/main.tcl
@@ -19,7 +19,7 @@ source $::launchdir/helpers.tcl
 package require cmdline
 package require Tclx
 
-set pidFile "/var/run/piaware.pid"
+set pidFile "$::env(SNAP_DATA)/piaware.pid"
 
 #
 # main - the main program
diff --git a/package/piaware.tcl b/package/piaware.tcl
index 2a16fef..e65b5f0 100644
--- a/package/piaware.tcl
+++ b/package/piaware.tcl
@@ -11,7 +11,11 @@ package require Itcl
 package require tryfinallyshim
 package require fa_sudo
 
-set piawarePidFile /var/run/piaware.pid
+if {[info exists ::env(SNAP_DATA)]} {
+	set piawarePidFile $::env(SNAP_DATA)/piaware.pid
+} else {
+	set piawarePidFile /var/run/piaware.pid
+}
 set piawareConfigFile /etc/piaware
 
 #
diff --git a/package/fa_sudo.tcl b/package/fa_sudo.tcl
index 9d97095..bfec0ff 100644
--- a/package/fa_sudo.tcl
+++ b/package/fa_sudo.tcl
@@ -42,6 +42,11 @@ namespace eval ::fa_sudo {
 	}
 
 	proc _make_droproot_args {user argv0 args} {
+		# do not droproot when snapped
+		if {[info exists ::env(SNAP)]} {
+			return [list {*}$args]
+		}
+
 		# try a few different things to drop root
 		set path [auto_execok $::fa_sudo::dropRootHelper]
 		if {$path ne ""} {
